Added release server publishing retry

fheinecke opened the pull request:
The release server publishing is failing due to a TLS issue. While it is being resolved I've added a retry to our pipelines to publish three times instead of once.

This should be removed once the issue is fixed.

Commit:
Added release server publishing retry

r0mant commented on .drone.yml:
```diff
@@ -16356,10 +16354,12 @@ steps:
-    docker run -i -v /tmpfs/creds:/tmpfs/creds \
-      -e DRONE_REPO -e DRONE_TAG -e RELCLI_BASE_URL -e RELCLI_CERT -e RELCLI_KEY \
-      $RELCLI_IMAGE auto_publish -f -v 6
+  - docker run -i -v /tmpfs/creds:/tmpfs/creds -e DRONE_REPO -e DRONE_TAG -e RELCLI_BASE_URL
+    -e RELCLI_CERT -e RELCLI_KEY $RELCLI_IMAGE auto_publish -f -v 6 || true
+  - docker run -i -v /tmpfs/creds:/tmpfs/creds -e DRONE_REPO -e DRONE_TAG -e RELCLI_BASE_URL
+    -e RELCLI_CERT -e RELCLI_KEY $RELCLI_IMAGE auto_publish -f -v 6 || true
+  - docker run -i -v /tmpfs/creds:/tmpfs/creds -e DRONE_REPO -e DRONE_TAG -e RELCLI_BASE_URL
+    -e RELCLI_CERT -e RELCLI_KEY $RELCLI_IMAGE auto_publish -f -v 6
```
Wouldn't this just always call the command 3 times? Can we instead wrap this in a "for" loop with a few iterations and abort it on success? Also, why 3? I think @logand22 said for him it worked on like 25th attempt, can we just make it retry 50 times or something?

logand22 replied:
It worked on second attempt for me. I mentioned 25 because I called the command many times after the first success to make sure it never occurred after the first success.

camscale replied:
I think all it needs is `|| \` on the end instead of `|| true`. Then it is one long pipeline that stops when one of the commands succeeds and fails when all fail.

camscale replied:
But if we do want to do 50 times or whatever, then a loop is a lot better. It shouldn't be that messy - let me knock something up.

camscale committed:
dronegen: Run auto_publish 10 times (from 3) in a loop

Change the drone generation to use a loop to run the `auto_publish`
relcli command instead of listing them one-by-one and loop 10 times
instead of 3. The loop will terminate the first time `relcli` succeeds.

The loop has an `|| false` at the end to ensure the loop command fails
if all invocations of `relcli` fail. With `set -e`, even though the exit
status of the loop is non-zero, the shell seems to continue. With the
`|| false` at the end, it makes it exit on failure. I'm not sure exactly
how drone runs the commands so this may not be necessary but it seems
safer.

e.g.

    set -e
    for i in $(seq 10); do false && break; done
    echo hello

This will echo "hello" even though all invocations inside the loop
failed.

    set -e
    for i in $(seq 10); do false && break; done || false
    echo hello

This will not echo "hello" - `set -e` causes an exit before that command
due to the `|| false`.
